src/compositions/useLspFactory.ts

```diff
@@ -1,12 +1,14 @@
-import { getSigner } from "@/services/provider.service";
+import { NETWORK_URL, PRIVATE_KEY } from "@/helpers/config";
+import { getState } from "@/stores";
 import { LSPFactory } from "@lukso/lsp-factory.js";
 
-export async function useLspFactory(): Promise<LSPFactory> {
-  const { provider } = await getSigner();
-  const { chainId } = await provider.getNetwork();
-  return new LSPFactory(provider as unknown as string, {
-    deployKey:
-      "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80",
-    chainId,
-  });
+export async function useLspFactory(): Promise<LSPFactory | null> {
+  const chainId = getState("chainId");
+  if (chainId && NETWORK_URL) {
+    return new LSPFactory(NETWORK_URL, {
```

Comment:
Actually let's initiate how docs suggest with dApp https://docs.lukso.tech/tools/lsp-factoryjs/getting-started#using-lspfactory-in-a-dapp, so pass only `window.ethereum` object if present. Let's leave passing controller key to deploy page where it can be specified by a user.

Reply:
Also please check how compositions are build, they should return individual functions so later we can add more methods to composition like `deploy`, for example:
```
export default function useLspFactory(): {
  getFactory: () => Promise<LSPFactory | null>;
  deploy: () => Promise<LSPFactory | null>
} {
  return {
    getFactory,
    deploy
  }
}
```